sw/device/lib/crypto/drivers/entropy.h

```diff
@@ -0,0 +1,139 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_DRIVERS_ENTROPY_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_DRIVERS_ENTROPY_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/status.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Seed material as specified in NIST SP 800-90Ar1 section 10.2.1.3.1. Up to 12
+ * words of seed material can be provided using this interface.
+ */
+typedef struct entropy_seed_material {
+  /**
+   * Number of words set in `data`. CSRNG will extend the `data` to zeros if the
+   * provided value is less than 12.
```

Comment:
nit: will CSRNG pad at the start or end? Can we just require this from the clients for the sake of simplicity or do you anticipate a considerable difference in performance?

Reply:
This is done by hardware at the moment. The padding is added at the end.